waiter/src/waiter/auth/authentication.clj

```diff
@@ -44,12 +44,15 @@
 (defn handle-request-auth
   "Invokes the given request-handler on the given request, adding the necessary
   auth headers on the way in, and the x-waiter-auth cookie on the way out."
-  [handler request user principal password]
-  (let [auth-params-map (auth-params-map principal user)
-        handler' (middleware/wrap-merge handler auth-params-map)]
-    (-> request
-        handler'
-        (add-cached-auth password principal))))
+  ([handler request principal password]
```

Comment:
Argument name is misleading, `principal ` should be `user`.

Reply:
I don't agree

Reply:
In the overloaded function call, `principal` becomes `user`, and `(auth-params-map principal)` becomes `principal`. I found that very confusing.

Reply:
there are 2 overloads, both call the 3rd one